File: 6.Two Pointer/python_solution/16-TwoSum.py
def two_sum(N,B,arr):
    left, right = 0, N - 1
    found = False
            
    while left < right:
      current_sum = arr[left] + arr[right]
                
      if current_sum == B:
        print(left, right)
        found = True
        break
      elif current_sum < B:
        left += 1
      else:
        right -= 1
            
    if not found:
      return "-1 -1"
      
N = 4
B = 9
arr = [2,7,11,15]
res = two_sum(N,B,arr)
print(res)

File: 6.Two Pointer/javascript_solution/16-TwoSum.js
function twoSum(N, B, arr) {
  let left = 0;
  let right = N - 1;
  let found = false;

  while (left < right) {
    const currentSum = arr[left] + arr[right];

    if (currentSum === B) {
      console.log(left, right);
      found = true;
      break;
    } else if (currentSum < B) {
      left++;
    } else {
      right--;
    }
  }

  if (!found) { 
    return "-1 -1";
  }
}

// Example usage:
const N = 4;
const B = 9;
const arr = [2, 7, 11, 15];

const res = twoSum(N, B, arr);
console.log(res);
